pkg/controller/utils/certs.go

```diff
@@ -74,28 +73,30 @@ func EnsureCertificateSecret(secretName string, secret *corev1.Secret, keyName s
 		)
 	}
 
+	operatorManaged, err := IsOperatorManaged(secret, certName)
```

Comment:
I understand from the other code, that if a cert is operator managed and invalid, we want to throw it away. That lgtm.

But what I do not understand is that if it is not operator managed we skip checking for dns names. Won't our components run into ussues if they run at  https://tigera-server.tigera-ns.svc.cluster.local, but don't have that subject (alt) name?

Reply:
We talked about this and I think you're right: I'll add some logging and set the controller degraded if the user-provided certs are not valid.

Reply:
Actually... the reason for reverting the cert DNS validation for custom certs is because we don't have a good way to detect that we have a hybrid cluster.